### PR TITLE
Replace Markdown training data in tests with YAML training data

### LINKED ISSUE #7737: Replace Markdown training data in tests with YAML training data
**Description of Problem**:
Support for Markdown is gonna be removed in Rasa Open Source 3.0. A lot of our unit tests are currently relying on training data format in Markdown format.

**Definition of Done**:
- [ ] All Markdown training data files are replaced with their equivalents in YAML (except if they are needed to test the Markdown readers / writers) 
- [ ] Unused Markdown files are deleted from the repository

### PR DESCRIPTION
**Pull request by alwx**
**Proposed changes**:
- fix #7737

There are still several markdown files in `data` directory.
- Files in `data/test_md` are used to test the conversion from MD to YAML;
- Files in `data/test/multiple_files_markdown`, `data/test/markdown_single_section` and `data/test_multifile_md_stories` are very MD-specific — they will be removed once the code that supports markdown and all the tests are removed.
- Files in `data/test_mixed_yaml_md_stories` are needed to check if you can mix MD and YAML in one project — these files can also be removed once we stop supporting MD.

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: tests/shared/core/training_data/story_reader/test_common_story_reader.py [hunk 1]
```diff
@@ -51,18 +47,9 @@ async def test_can_read_test_story(stories_file: Text, default_domain: Domain):
     assert tracker.events[4] == ActionExecuted("action_listen")
 
 
-@pytest.mark.parametrize(
-    "stories_file",
-    [
-        "data/test_stories/stories_checkpoint_after_or.md",
```
Comment (joejuzl) [inline tests/shared/core/training_data/story_reader/test_common_story_reader.py:57]: Is it ok to stop testing this with markdown?

File: tests/shared/nlu/training_data/test_training_data.py [hunk 2]
```diff
@@ -588,7 +597,7 @@ def test_training_data_conversion(
             rasa.shared.data.yaml_file_extension(),
         ),
         ("data/examples", rasa.shared.data.yaml_file_extension()),
-        ("data/examples/rasa/demo-rasa.md", rasa.shared.data.markdown_file_extension()),
+        ("data/examples/rasa/demo-rasa.yml", rasa.shared.data.yaml_file_extension()),
```
Comment (joejuzl) [inline tests/shared/nlu/training_data/test_training_data.py:600]: I guess we still want to test `markdown_file_extension` here.

File: tests/shared/core/training_data/story_reader/test_common_story_reader.py [hunk 3]
```diff
@@ -71,15 +58,10 @@ async def test_can_read_test_story_with_checkpoint_after_or(
     assert len(trackers) == 2
 
 
-@pytest.mark.parametrize(
-    "stories_file",
-    [
-        "data/test_stories/stories_with_cycle.md",
```
Comment (joejuzl) [inline tests/shared/core/training_data/story_reader/test_common_story_reader.py:77]: likewise for all the others in this file.

**Review by joejuzl — CHANGES_REQUESTED** [review]

File: tests/shared/core/training_data/story_reader/test_common_story_reader.py (hunk 1 above)
Comment (alwx) [reply tests/shared/core/training_data/story_reader/test_common_story_reader.py:57]: good catch, returned it back

**Review by joejuzl — APPROVED** [review]

File: data/test/lookup_tables/lookup_table.yml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+    - I want to grab [tacos](food)
+    - I am searching for a [pizza](food) spot
+    - I would like to drink [sweet berry wine](beverage) with my meal
+- lookup: plates
+  examples: |
+    - tacos
+    - beef
+    - mapo tofu
+    - burrito
+    - lettuce wrap
```
Comment (wochinge) [inline data/test/lookup_tables/lookup_table.yml:15]:
```suggestion
    - lettuce wrap

```

File: data/test_evaluations/end_to_end_trips_circuit_breaker.yml [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+  - action: utter_greet
+  - action: utter_greet
+  - action: utter_greet
+  - action: utter_greet
+  - action: utter_greet
+  - action: utter_greet
+  - action: utter_greet
+  - action: utter_greet
+  - action: utter_greet
+  - action: utter_greet
```
Comment (wochinge) [inline data/test_evaluations/end_to_end_trips_circuit_breaker.yml:16]:
```suggestion
  - action: utter_greet

```

File: data/test_evaluations/stories_trip_circuit_breaker.yml [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+  - action: utter_greet
+  - action: utter_greet
+  - action: utter_greet
+  - action: utter_greet
+  - action: utter_greet
+  - action: utter_greet
+  - action: utter_greet
+  - action: utter_greet
+  - action: utter_greet
+  - action: utter_greet
```
Comment (wochinge) [inline data/test_evaluations/stories_trip_circuit_breaker.yml:15]:
```suggestion
  - action: utter_greet

```

File: data/test_evaluations/story_unknown_entity.yml [hunk 7]
```diff
@@ -0,0 +1,6 @@
+stories:
+- story: simple_story_with_unknown_entity
+  steps:
+  - user: |
+      I am [rasa](name)
+    intent: utter_greet
```
Comment (wochinge) [inline data/test_evaluations/story_unknown_entity.yml:6]:
```suggestion
    intent: utter_greet

```

**Review by wochinge — COMMENTED** [review]
Nice work 💯 

Do we still need all md story files in `data/test_stories`? E.g. [this test](https://github.com/RasaHQ/rasa/blob/78fdc5dac0774589229b758c6b1cf51ee41d5bf7/tests/test_server.py#L535) still uses one of the MD files which in my opinion doesn't make sense.

**Comment by alwx** [comment]
We cannot remove all the stuff from there because of several reasons:
1. These files are used in `test_markdown_story_reader.py` — but we can easily remove tests once we drop the support of the reader
2. Some of them are used in `test_common_story_reader.py` which also makes sense because we still support the markdown.

I will remove other usages though.

**Pull request merged**